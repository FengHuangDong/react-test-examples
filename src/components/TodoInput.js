import React, { Component } from 'react'
import { Field, reduxForm } from 'redux-form'

export class TodoInput extends Component {

  render() {
    const { handleSubmit } = this.props
    return (
      <div className="form-group row">
        <div className="col-sm-10">
          <form className="form-inline" onSubmit={ handleSubmit }>
            <Field className="form-control"  placeholder="add todo here" name="title" component="input" type="text" />
            <button style={{marginLeft:"10px"}} className="btn btn-success" type="submit">Submit</button>
          </form>
        </div>
      </div>
    )
  }
}


export default reduxForm({
  // a unique name for the form
  form: 'todo'
})(TodoInput);
