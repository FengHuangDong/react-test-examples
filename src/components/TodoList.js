import React, { Component } from 'react'
import PropTypes from 'prop-types';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as todoActions from '../actions/todo';
import { SHOW_ALL, SHOW_COMPLETED, SHOW_ACTIVE } from "../actions/actionsTypes";

export class TodoList extends Component {
  render () {
    return (
      <React.Fragment>
        <nav style={{ marginTop: "60px" }}>
          <ol className="breadcrumb">
            <li
              className={"breadcrumb-item "+ (this.props.visibilityFilter === SHOW_ALL ? 'active' : '') }
              onClick={() => this.props.actions.setVisibilityFilter(SHOW_ALL)}
            >
             All
            </li>
            <li
               className={"breadcrumb-item "+ (this.props.visibilityFilter === SHOW_COMPLETED ? 'active' : '') }
              onClick={() => this.props.actions.setVisibilityFilter(SHOW_COMPLETED)}
            >
              Completed
            </li>
            <li
               className={"breadcrumb-item "+ (this.props.visibilityFilter === SHOW_ACTIVE ? 'active' : '') }
              onClick={() => this.props.actions.setVisibilityFilter(SHOW_ACTIVE)}
            >
              Active
            </li>
          </ol>
        </nav>
        {
          this.props.todos.length !== 0 ? (
            <table
              style={{ marginTop: "60px" }}
              className="table table-hover table-dark"
            >
              <thead>
                <tr>
                  <th scope="col">Todos</th>
                  <th scope="col">Actions</th>
                </tr>
              </thead>
              <tbody>
                {this.props.todos.map((todo,i) => (
                  <tr key={i}>
                    <td
                      style={{
                        textDecoration: todo.completed ? "line-through" : "none"
                      }}
                    >
                      {todo.title} {todo.completed === true ? "(completed)" : ""}
                    </td>
                    <td>
                      <span
                        className="fas fa-minus-circle"
                        onClick={() => this.props.actions.removeFromList(todo)}
                        style={{
                          color: "white",
                          fontSize: "20pt",
                          marginRight: "20px"
                        }}
                      />
                      <span
                        className="fas fa-check-circle"
                        onClick={() => this.props.actions.toogleTodo(todo.id)}
                        style={{ color: "white", fontSize: "20pt" }}
                      />
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
          ) : (
            <div
              style={{ marginTop: "50px" }}
              className="col-lg-10 col-md-10 col-xs-12 col-sm-12 offset-lg-1"
            >
              <div className="alert alert-danger" role="alert">
                Todo List is empty or Filter results show no results
              </div>
            </div>
          )
        }
      </React.Fragment>
    )
  }
}

TodoList.propTypes = {
  todos: PropTypes.array
}
export const getVisibleTodos = (todos, filter) => {
  switch (filter) {
    case SHOW_ALL:
      return todos;
    case SHOW_COMPLETED:
      return todos.filter(t => t.completed);
    case SHOW_ACTIVE:
      return todos.filter(t => !t.completed);
    default:
      throw new Error("Unknown filter: " + filter);
  }
};

export function mapStateToProps(state, props) {
    return {
      todos: getVisibleTodos(state.todos, state.visibilityFilter),
      visibilityFilter: state.visibilityFilter
    }
}

export function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(todoActions, dispatch)
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
