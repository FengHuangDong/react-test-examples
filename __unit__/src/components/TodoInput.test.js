import React from 'react';
import {TodoInput} from "../../../src/components/TodoInput"
import {shallow} from 'enzyme'
import toJson from "enzyme-to-json"


describe('TodoInput Components', () => {
  let mockProps,wrapper
  beforeEach(() => {
    mockProps = {
      handleSubmit:jest.fn()
    }
    wrapper = shallow(<TodoInput {...mockProps} />)
  })
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should submit handle to form ', () => {
    wrapper.find("form").simulate('submit')
    expect(mockProps.handleSubmit).toHaveBeenCalled();
  });
})