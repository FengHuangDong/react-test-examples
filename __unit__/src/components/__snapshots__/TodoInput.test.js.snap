// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput Components should render component 1`] = `
<div
  className="form-group row"
>
  <div
    className="col-sm-10"
  >
    <form
      className="form-inline"
      onSubmit={[MockFunction]}
    >
      <Field
        className="form-control"
        component="input"
        name="title"
        placeholder="add todo here"
        type="text"
      />
      <button
        className="btn btn-success"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
